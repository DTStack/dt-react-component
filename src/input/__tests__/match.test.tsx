import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import Match from '../match';

describe('Test Input.Match component', () => {
    it('Should match snapshot', () => {
        const { asFragment } = render(<Match />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Should support searchType', () => {
        const fn = jest.fn();
        const { rerender, container } = render(<Match onTypeChange={fn} />);
        const elements = container.querySelectorAll('svg.dtc-input-match-suffixItem')!;
        expect(elements).toHaveLength(4);

        fireEvent.click(elements[0]);

        expect(fn).toBeCalledWith('caseSensitive');

        rerender(<Match searchType="caseSensitive" onTypeChange={fn} />);
        expect(container.querySelectorAll('svg.dtc-input-match-suffixItem')!).toHaveLength(4);
        expect(container.querySelectorAll('svg.dtc-input-match-suffixItem')[0].classList).toContain(
            'dtc-input-match-suffixItem__active'
        );
        fireEvent.click(container.querySelectorAll('svg.dtc-input-match-suffixItem')[0]);
        expect(fn).toBeCalledWith('fuzzy');
    });

    it('Should support filterOptions', () => {
        const { container } = render(<Match filterOptions={['tail']} />);

        expect(container.querySelectorAll('svg.dtc-input-match-suffixItem')!).toHaveLength(1);
    });

    it('Should support onSearch and onPressEnter', () => {
        const searchFn = jest.fn();
        const pressEnterFn = jest.fn();
        const { container } = render(
            <Match
                value="test"
                searchType="front"
                onSearch={searchFn}
                onPressEnter={pressEnterFn}
            />
        );

        fireEvent.keyDown(container.querySelector('input')!, { key: 'Enter', charCode: 13 });

        expect(searchFn).toBeCalledWith('test', 'front');
        expect(pressEnterFn).toBeCalled();
    });

    it('Should support get value', () => {
        const fn = jest.fn();
        const { container } = render(<Match value="test" onChange={fn} />);

        const inputEl = container.querySelector('input')!;
        fireEvent.change(inputEl, { target: { value: 'test1' } });

        expect(fn).toBeCalled();
    });
});
