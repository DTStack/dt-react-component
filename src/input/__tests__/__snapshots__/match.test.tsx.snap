// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input.Match component Should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="ant-input-affix-wrapper dtc-input-match"
  >
    <input
      class="ant-input"
      type="text"
      value=""
    />
    <span
      class="ant-input-suffix"
    >
      <div
        class="dtc-input-match-items"
      >
        <span
          class="dtc-input-match-suffixItem"
          data-mock-icon="CaseSensitiveMatchOutlined"
        />
        <span
          class="dtc-input-match-suffixItem"
          data-mock-icon="ExactMatchOutlined"
        />
        <span
          class="dtc-input-match-suffixItem"
          data-mock-icon="PrefixMatchOutlined"
        />
        <span
          class="dtc-input-match-suffixItem"
          data-mock-icon="SuffixMatchOutlined"
        />
      </div>
    </span>
  </span>
</DocumentFragment>
`;
